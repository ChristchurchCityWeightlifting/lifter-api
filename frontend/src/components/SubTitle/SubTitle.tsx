/** @format */

import React from "react";
import { Typography, TypographyProps } from "@mui/material";

interface SubTitleProps extends TypographyProps {
  children?: React.ReactNode;
}

const SubTitle: React.FC<React.PropsWithChildren<SubTitleProps>> = (
  props: SubTitleProps
) => {
  return (
    <Typography color="primary" variant="subtitle2" {...props}>
      {props.children}
    </Typography>
  );
};

export default SubTitle;
