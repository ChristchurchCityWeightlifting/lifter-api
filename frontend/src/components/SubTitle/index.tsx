/** @format */

export { default } from "./SubTitle";
