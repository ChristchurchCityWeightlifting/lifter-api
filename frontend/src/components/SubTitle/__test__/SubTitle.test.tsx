/** @format */

import React from "react";
import { describe, test, expect } from "@jest/globals";
import renderer from "react-test-renderer";
import SubTitle from "../SubTitle";

describe("SubTitle components tests", () => {
  test("renders without text", () => {
    const tree = renderer.create(<SubTitle></SubTitle>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("renders with text", () => {
    const tree = renderer.create(<SubTitle>Test Text</SubTitle>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("renders with text and props", () => {
    const tree = renderer
      .create(
        <SubTitle sx={{ textDecoration: "underline" }}>Test Text</SubTitle>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
