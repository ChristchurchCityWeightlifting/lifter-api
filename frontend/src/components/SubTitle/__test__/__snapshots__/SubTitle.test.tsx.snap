// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubTitle components tests renders with text 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-a2zpya-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:500;font-size:0.875rem;line-height:1.57;letter-spacing:0.00714em;color:#1976d2;}",
      }
    }
    data-emotion="css a2zpya-MuiTypography-root"
  />,
  <h6
    className="MuiTypography-root MuiTypography-subtitle2 css-a2zpya-MuiTypography-root"
  >
    Test Text
  </h6>,
]
`;

exports[`SubTitle components tests renders with text and props 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1if87dc-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:500;font-size:0.875rem;line-height:1.57;letter-spacing:0.00714em;color:#1976d2;-webkit-text-decoration:underline;text-decoration:underline;}",
      }
    }
    data-emotion="css 1if87dc-MuiTypography-root"
  />,
  <h6
    className="MuiTypography-root MuiTypography-subtitle2 css-1if87dc-MuiTypography-root"
  >
    Test Text
  </h6>,
]
`;

exports[`SubTitle components tests renders without text 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-a2zpya-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:500;font-size:0.875rem;line-height:1.57;letter-spacing:0.00714em;color:#1976d2;}",
      }
    }
    data-emotion="css a2zpya-MuiTypography-root"
  />,
  <h6
    className="MuiTypography-root MuiTypography-subtitle2 css-a2zpya-MuiTypography-root"
  />,
]
`;
